### PR TITLE
Remove mpi and batch dictionaries

### PR DESCRIPTION
**Pull request by douglasjacobsen**
This merge removes the two primary dictionaries in the ramble.yaml file `ramble:mpi` and `ramble:batch` in favor of using the variables `mpi_command` and `batch_submit` respectively.

Tests and documenatation is updated to reflect this too.

### CODE REVIEW

**Comment by douglasjacobsen** [comment]
@rfbgo  and @dodecatheon I'd like you to both review this, since it's a pretty major breaking change relative to the previous way this worked. I also think after this gets merged, we should make a new tag.

**Review by dodecatheon — APPROVED** [review]
LGTM

File: lib/ramble/ramble/schema/workspace.py
Comment (rfbgo) [inline lib/ramble/ramble/schema/workspace.py:68]: I think it would be nice if we can temporarily keep some code in that catches when a user is using the "old style" and warn them 

If I understood the PR correctly there is a chance where they end up with the default values for the variables and get a bit confused about what is going on

**Review by rfbgo — CHANGES_REQUESTED** [review]
LGTM, but I wonder if we should do a "software transition" and help the users out a bit more?

File: lib/ramble/ramble/schema/workspace.py
Comment (douglasjacobsen) [reply lib/ramble/ramble/schema/workspace.py:68]: OK, I've added a deprecation check now after our offline discussion. Let me know if this seems to help.

Comment (rfbgo) [reply lib/ramble/ramble/schema/workspace.py:68]: Looks go to me. What do you see as the path to deprecating things in the future, and does it include `@deprecated`? If so it might make sense to use it here too?

Comment (douglasjacobsen) [reply lib/ramble/ramble/schema/workspace.py:68]: My current plan has been to mark things as deprecated in the code, and use `tty.warn` to throw a warning to the user.

I'm then going to make tasks to review these in a few releases, so we can remove the deprecated functionality before 1.0 is released.

Where possible, we can use `@deprecated`, but some of this type of stuff won't be nice with that (since we're not deprecating a function, we're really deprecating part of a function).

But maybe we should push these things into functions more clearly so we can use the deprecated decorator. What are your thoughts?

Comment (rfbgo) [reply lib/ramble/ramble/schema/workspace.py:68]: My only "requirement" for this PR is that we help the user know when they are using functionality that is going away, and give them a smooth transition, I think you've already achieved this :) 

I like everything you propose, and see this as a good chance to kick the tires on whatever we think our post v1.0 deprecation process is (I don't really mind what that looks like, but I too like the idea of a function level decorator that calls `tty.warn` so pulling the deprecated portion into functions sounds very nice and tidy to me)

Up to you if you want to capture that here or later

Comment (douglasjacobsen) [reply lib/ramble/ramble/schema/workspace.py:68]: Spending some time looking into how deprecate works, I don't think it's going to end up being any better than what I've implemented currently.

So, I think the following are kind of my current thinking as far as how we should go about handling deprecated thing:
1) Make an obvious 'warning' print in the code to let people know their functionality is deprecated. I don't really love the way this is handled by python currently, because it's printed when the functionality is used, rather than warning at the end of execution. But maybe we can explore ways to fix that outside of this PR. Ultimately this is done with `tty.warn` calls.
2) Mark the code with `# DEPRECATED` and list something that describes when it can be removed
3) Make a task tracking when we will remove the functionality.

Comment (rfbgo) [reply lib/ramble/ramble/schema/workspace.py:68]: Yea, I agree. Lets merge this and possibly follow up with another task when needed?

Comment (douglasjacobsen) [reply lib/ramble/ramble/schema/workspace.py:68]: SGTM

**Review by rfbgo — APPROVED** [review]

**Pull request merged**
